core/chains/evm/types/address.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"bytes"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	commontypes "github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+var _ commontypes.Hashable = (*Address)(nil)
+
+type Address struct {
+	commontypes.Hashable
+	nativeAddress common.Address
+}
+
+func (a *Address) MarshalText() (text []byte, err error) {
+	return a.nativeAddress.Bytes(), nil
+}
+
+func (a *Address) UnmarshalText(text []byte) error {
+	a.nativeAddress = common.BytesToAddress(text)
+	return nil
+}
+
+func (a *Address) String() string {
+	return a.nativeAddress.String()
+}
+
+func (a *Address) Equals(h *Address) bool {
+	return bytes.Equal(a.nativeAddress.Bytes(), h.nativeAddress.Bytes())
+}
+
+func (a *Address) Empty() bool {
+	return a == nil || a.nativeAddress == common.Address{}
```

Comment:
I added empty because there were some places which check if we are sending to zeroaddress etc.
Ideally we can remove it, and also the code that needs it.
As of now, I don't know what well-defined terms could be on different chains.

Reply:
I mean well-defined go terms, like `nil` or `zero`. Otherwise, some kind of opposite term might be more intuitive and idiomatic, like how `IsValid` is used in `package reflect` (in addition `IsNil` and `IsZero`):
https://pkg.go.dev/reflect#Value.IsValid
> IsValid reports whether v represents a value. It returns false if v is the zero Value...

Reply:
In this case, it was ethereum's ZeroAddress. One place in txmgr wanted to block sending funds to a zero address.
This could be a valid use-case for some chains. I think even in Eth, you could destroy a smartcontract by some zeroaddress related tweak, IIRC.

Don't have enough data from all chains to say what's the right use of this term.
Frankly, I am more in favor of removing that check from txmgr, and then from this interface in a later PR.